### DISCUSSION TITLE
Mark unexisting tasks as complete when clear a dag execution

### DISCUSSION THREAD

**Discussion by alejandrofm** [apache/airflow#35730] [General]
### Description

Hi! When reprocessing data, there are a lot of cases where this happens, for example when need to reprocess it will create tasks where previously there weren't any.
I want to be able to mark those tasks as success or better, let them stay on "none" unless when clearing the dag I explicitly tell it to add non-existent tasks to the moment on the dag execution.
![image](https://github.com/apache/airflow/assets/19591174/7bb92ad2-e7da-47f7-8410-55677207f485)

### Use case/motivation

Now I have to do it manually when each of the tasks goes from scheduled to running, some of these processes run serially, so I need to be in front of Airflow constantly while reprocessing making sure that this particular task would non run.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
This is part of DAG versioning https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-36+DAG+Versioning that we don't even work on yet. Currently Airflow always runs the current version of the code - we have no snapshots of the code that was used when it was run in the past. So this as a feature will not be implemented separately. 

currently if you want to avoid that - you should create a new version of the DAG with different DAG id if you want to change the structure of the dag significantly (add or remove many tasks).

Converting it to discussion if more nseded.
